### PR TITLE
Add support for Dencun hardfork

### PR DESCRIPTION
**Pull request by k4m4**
- Update ethereumjs dependencies
- Modify VerifyingProvider to be compatible with ethereumjs@latest
- Update tests

### CODE REVIEW

File: src/provider.ts [hunk 1]
```diff
@@ -70,15 +72,34 @@ export class VerifyingProvider {
...
-    this.common = new Common({
-      chain,
-      hardfork: chain === Chain.Mainnet ? Hardfork.Shanghai : undefined,
-    });
     const _blockNumber = BigInt(blockNumber);
     this.latestBlockNumber = _blockNumber;
     this.blockHashes[bigIntToHex(_blockNumber)] = blockHash;
+    this.common = new Common({
+      chain,
+      hardfork: chain === Chain.Mainnet ? Hardfork.Cancun : undefined,
```
Comment (shresthagrawal) [inline src/provider.ts:83]: Why not always set the hard fork to `Cancun` instead of setting it to undefined?

File: src/provider.ts [hunk 2]
```diff
@@ -653,49 +674,51 @@ export class VerifyingProvider {
...
       balance: BigInt(proof.balance),
       storageRoot: proof.storageHash,
       codeHash: proof.codeHash,
     });
-    const isAccountValid = account
-      .serialize()
-      .equals(expectedAccountRLP ? expectedAccountRLP : emptyAccountSerialize);
+    const isAccountValid = equalsBytes(
+      account.serialize(),
+      expectedAccountRLP ? expectedAccountRLP : emptyAccountSerialize,
```
Comment (shresthagrawal) [inline src/provider.ts:703]:
```suggestion
      expectedAccountRLP || emptyAccountSerialize,
```

File: src/provider.ts (hunk 1 above)
Comment (k4m4) [reply src/provider.ts:83]:
```suggestion
      hardfork: Hardfork.Cancun,
```

**Comment by shresthagrawal** [comment]
LGTM

**Pull request merged**